src/ui/flutter_app/lib/engine/native_engine.dart

```diff
@@ -20,41 +20,43 @@ import 'dart:async';
 
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
-import 'package:sanmill/common/config.dart';
 import 'package:sanmill/mill/position.dart';
 import 'package:sanmill/mill/types.dart';
+import 'package:sanmill/shared/common/config.dart';
 
 import 'engine.dart';
 
 class NativeEngine extends Engine {
-  static const platform = const MethodChannel('com.calcitem.sanmill/engine');
+  static const platform = MethodChannel('com.calcitem.sanmill/engine');
   bool isActive = false;
 
+  @override
   Future<void> startup() async {
     await platform.invokeMethod('startup');
-    await waitResponse(['uciok'], sleep: 100, times: 0);
+    await waitResponse(['uciok']);
   }
 
   Future<void> send(String command) async {
-    print("[engine] send: $command");
+    debugPrint("[engine] send: $command");
     await platform.invokeMethod('send', command);
   }
 
   Future<String?> read() async {
-    return await platform.invokeMethod('read');
+    return platform.invokeMethod('read');
   }
 
+  @override
   Future<void> shutdown() async {
     isActive = false;
     await platform.invokeMethod('shutdown');
   }
 
   Future<bool?> isReady() async {
-    return await platform.invokeMethod('isReady');
+    return platform.invokeMethod('isReady');
   }
 
-  FutureOr<bool> isThinking() async {
-    return await platform.invokeMethod('isThinking');
+  Future<bool> isThinking() async {
+    return platform.invokeMethod<bool>('isThinking') as Future<bool>;
```

Comment:
Need to revert or fix. Otherwise, throw an exception when taping the board. (Human Vs. AI)